github/prci.py

```diff
@@ -19,11 +22,76 @@
 )
 from internals.gql import util, queries
 
-
 logger = logging.getLogger(__name__)
 
 
 ERROR_BACKOFF_TIME = 600
+OS_PID_FILE_PATH = "/run/"
+OS_PID_FILE = os.path.join(OS_PID_FILE_PATH, "prci.pid")
+
+
+def pid_file_creation():
+    """
+        Creates PID file with Proccess ID as content.
+    """
+    with open(OS_PID_FILE, "w") as pid_file:
```

Comment:
Creating a PID file is a good practice for and recommended for POSIX Style applications.

PID file is used to record process ID at the beginning of the program, as the PID will never change during execution, this is why is a write once situation.

Is a good practice to ensure that the prci was not running before and take the right decision. Also, this introduce a interface for future expansions in prci.

Reply:
Shouldn't this be done apart of the pruning mechanism? And if the file already exists it's not being overwritten by a second execution. It will contain the previous PID.

Reply:
You are mixing concerns here. PID and prune of the process are a different process in essence.

PID is used in all Unix for mechanisms like concurrency and locking. It introduces a simple way to handle the process that is running, and in future, can be improved to ensure that only one ( or extend the execution to more than one) process.

Pruning is a requirement for prci application, PID is to introduce a more fined grained execution control mechanism.

Reply:
They are implemented in the same function, that's the issue.